### PR TITLE
Params expansion 150

### PR DESCRIPTION
**Pull request by ameynert**
Addresses issue #150 .

Adds `seq_center` optional column to samplesheet.csv to be passed to STARsolo subworkflow: overrides `params.seq_center` if specified.

Adds `expected_cells` optional column to samplesheet.csv to be passed to STARsolo, CellRanger, and Alevin subworkflows.

Note this PR is going in now to have work to date reviewed - still waiting on a bugfix bioconda release 0.5.3 of simpleaf to test the Alevin implementation.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -38,14 +38,24 @@ TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,
...
+| `fastq_2`        | Full path to FastQ file for Illumina short reads 2. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".                                                                                                                                                                                      |
+| `expected_cells` | Number of cells expected for a sample. Must be an integer. If multiple rows are provided for the same sample, this must be the same number for all rows, i.e. the total number of expected cells for the sample.                                                                                                |
+| `seq_center`     | Sequencing center for the sample. If multiple rows are provided for the same sample, this must be the same string for all rows. Samples sequenced at different centers are considered different samples and must have different identifiers. Used for STARsolo BAM outputs only. Overrides `params.seq_center`. |
```
Comment (jfy133) [inline docs/usage.md:47]: Are these optional? description above and in `expected cells` below sounds like they are so maybe indicate optional in the table

File: subworkflows/local/input_check.nf [hunk 2]
```diff
@@ -27,8 +28,10 @@ workflow INPUT_CHECK {
 def create_fastq_channel(LinkedHashMap row) {
     // create meta map
     def meta = [:]
-    meta.id         = row.sample
-    meta.single_end = row.single_end.toBoolean()
+    meta.id             = row.sample
+    meta.single_end     = row.single_end.toBoolean()
+    meta.expected_cells = row.expected_cells != null ? row.expected_cells : null
+    meta.seq_center     = row.seq_center ? row.seq_center : params.seq_center
```
Comment (jfy133) [inline subworkflows/local/input_check.nf:34]: Just to check, seq_center is not reuired in the samplesheet (given it's inserted intothe BAM header?)

**Review by jfy133 — APPROVED** [review]
Overall LGTM, just some minor questions

File: subworkflows/local/input_check.nf (hunk 2 above)
Comment (ameynert) [reply subworkflows/local/input_check.nf:34]: It's not required in the samplesheet. It's an optional annotation in the RG line in the BAM header.

File: docs/usage.md (hunk 1 above)
Comment (ameynert) [reply docs/usage.md]: New commit addresses this.

**Review by apeltzer — APPROVED** [review]

**Pull request merged**